recipes/hidapi/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+
+
+class HidapiConan(ConanFile):
+    name = "hidapi"
+    description = "HIDAPI is a multi-platform library which allows an application to interface " \
+                  "with USB and Bluetooth HID-Class devices on Windows, Linux, FreeBSD, and macOS."
+    topics = ("libusb", "hid-class", "bluetooth")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/libusb/hidapi"
+    license = "GPL-3-or-later", "BSD-3-Clause"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+    }
+    generators = "pkg_config"
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def build_requirements(self):
+        if self.settings.compiler != "Visual Studio":
+            self.build_requires("libtool/2.4.6")
+
+    def requirements(self):
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("libusb/1.0.24")
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "configure.ac"),
+                              "AC_CONFIG_MACRO_DIR", "dnl AC_CONFIG_MACRO_DIR")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        with tools.chdir(self._source_subfolder):
+            self.run("./bootstrap")
+        if self.settings.os == "Macos":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "configure"),
+                                  r"-install_name \$rpath/", "-install_name ")
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        args = ["--enable-shared" if self.options.shared else "--disable-shared",
+                "--enable-static" if not self.options.shared else "--disable-static"]
+        self._autotools.configure(configure_dir=self._source_subfolder, args=args)
+        return self._autotools
+
+    def build(self):
+        self._patch_sources()
+        if self.settings.compiler == "Visual Studio":
+            self._build_msvc()
+        else:
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def _build_msvc(self):
+        msbuild = MSBuild(self)
+        msbuild.build(os.path.join(self._source_subfolder, "windows", "hidapi.sln"),
+                      platforms={"x86": "Win32"})
```

Comment:
Does the visual studio solution build both static and shared libraries?

Reply:
Not, it build shared library only, should `shared: False` be deleted during `config_options` for Visual Studio?

Reply:
🥵Seams CMake building is on the way: https://github.com/libusb/hidapi/pull/220, Hope life become easier later in future.

Reply:
I would do the following:

```python
def config_options(self):
    if self.settings.compiler == "Visual Studio":
        self.options["shared"] = True
def configure(self):
    if not self.options.shared:
        raise ConanInvalidConfiguration("Static libraries are not available")
```
Removing the option would make it ambiguous.
This is documentation by code ™️  🤣 .

Untested + don't know whether the hooks will accept this.

Reply:
Thanks, already test and push.